src/planning/task_planner/scripts/task_planner.py

```diff
@@ -34,63 +36,65 @@
 import os
 import json
 from enum import Enum
+import pyrebase
+import time
 
 from utils import get_quaternion
 
 class TaskPlanner:
     def __init__(self):
         rospy.init_node('task_planner')
-
-        self.wakeword_topic_name = rospy.get_param("wakeword_topic_name",
-                             "/interface/wakeword_detector/wakeword")
-        
         rospack = rospkg.RosPack()
         base_dir = rospack.get_path('task_planner')
         locations_file = rospy.get_param("locations_file", "config/locations.json")
         locations_path = os.path.join(base_dir, locations_file)
-
         self.goal_locations = json.load(open(locations_path))
+
+        firebase_secrets_path = os.path.expanduser("~/firebasesecrets.json")
+        if not os.path.isfile(firebase_secrets_path):
+            raise FileNotFoundError("Firebase secrets file not found")
+        
+        with open(firebase_secrets_path, 'r') as f:
+            config = json.load(f)
+
+        with open(os.path.expanduser("~/alfred-autonomy/src/planning/task_planner/config/firebase_schema.json")) as f:
+            self.state_dict = json.load(f)
+
+        self.firebase = pyrebase.initialize_app(config)
+        self.db = self.firebase.database()
+        self.db.remove("")
+        self.state_dict["button_callback"] = False
+        self.db.set(self.state_dict) # fill
+ 
+        self.last_update = time.time()
+
+        self.bot_state = BotStateManager(self.stateUpdateCallback)
```

Comment:
botstatemanager and firebase have to somehow be merged to ensure teleop runs smoothly. architecture to be discussed